### PR TITLE
Adjustments for 5.4.2

### PR DESCRIPTION
**Pull request by joshmoore**
Various last-minute adjustments to the docs for the 5.4.2 release.

cc: @hflynn @pwalczysko @kennethgillen @jburel

### CODE REVIEW

**Comment by joshmoore** [comment]
Note: I'm concerned about the naming conflict between `public` and `publication`.

File: omero/index.rst
Comment (jburel) [inline omero/index.rst]: maybe also mention latest web apps?

Comment (joshmoore) [reply omero/index.rst]: Pushed. Thoughts? Also, would be quite nice to see _some_ public data on demo. cc: @pwalczysko

**Comment by joshmoore** [comment]
Going to hold off on further `public(action).rst` changes until we've had a chance to discuss.

File: omero/sysadmins/public.rst
Comment (jburel) [inline omero/sysadmins/public.rst:152]: update link to http://rsob.royalsocietypublishing.org/content/7/11/170099

File: omero/sysadmins/public.rst
Comment (jburel) [inline omero/sysadmins/public.rst]: better to use ``how the production server in Dundee`` was ...

File: omero/sysadmins/public.rst
Comment (jburel) [inline omero/sysadmins/public.rst]: dot missing at the end

File: omero/sysadmins/public.rst
Comment (joshmoore) [reply omero/sysadmins/public.rst:152]: Isn't this precisely what DOIs are for?

File: omero/sysadmins/public.rst
Comment (joshmoore) [reply omero/sysadmins/public.rst]: I purposefully used "nightshade" because otherwise it isn't clear what playbooks to read in that repo. Otherwise, we need to rename the files.

Comment (jburel) [reply omero/sysadmins/public.rst]: in that case maybe the suggested text and in brackets or i.e. "nightshade"

File: omero/sysadmins/public.rst
Comment (jburel) [reply omero/sysadmins/public.rst:152]: it is but elsewhere in the doc we use the "final" link
unifying makes the review easier

Comment (joshmoore) [reply omero/sysadmins/public.rst:152]: I can't get that to work. Minimally those URLs use different subpaths: `/content/royopenbio` for jpegs vs. just `/content` for the target of the DOI.

File: omero/sysadmins/public.rst
Comment (joshmoore) [reply omero/sysadmins/public.rst]: Updated following line to have a period.

File: omero/sysadmins/public.rst
Comment (jburel) [reply omero/sysadmins/public.rst:152]: ignore in that case

Comment (joshmoore) [reply omero/sysadmins/public.rst:152]: Looking for a "component DOI" now: http://blogs.plos.org/mfenner/2011/03/26/direct-links-to-figures-and-tables-using-component-dois/

File: omero/index.rst
Comment (pwalczysko) [inline omero/index.rst:36]: I know that the target site for this link was not dealt with in this PR, but
screen shot 2018-01-19 at 10 18 26
looks strange. Is this intended ?

Comment (joshmoore) [reply omero/index.rst:36]: I think that's mostly if not completely a result of the `latest/` redirect.

File: omero/sysadmins/index.rst
Comment (pwalczysko) [inline omero/sysadmins/index.rst:146]: screen shot 2018-01-19 at 10 35 13
This seems tp be a strange order. First I am told about 2 items which are (at least highlighted as) particular, only then about an item which claims to be "complete workflow".

Comment (joshmoore) [reply omero/sysadmins/index.rst:146]: I went back and forth on this quite a bit. The introduction to the workflow, though, links to public_user making it something of a prerequisite. Additionally, "complete workflow" doesn't show everything in the other two sections. So options I can think of:

 * Integrate "complete workflow" into the rest of the page using the single example over and over again (high cost, but I think valuable)
 * Append "Setup:" or "Howto:" to the preliminary items
 * Other names for "Complete workflow"
    - "Full example for ..."
    - "Motivation: ..."

File: omero/sysadmins/public.rst
Comment (will-moore) [inline omero/sysadmins/public.rst:159]: Do we want to indicate which playbooks we're referring to here, since there are a few at that repo?

Comment (joshmoore) [reply omero/sysadmins/public.rst:159]: There are several files that are related. If we're ok with expanding this section substantially, then can do. cc: @kennethgillen @manics

**Comment by pwalczysko** [comment]
Not sure about the capitalization or not or the ``First Name, Last Name, username and password`` in this line https://github.com/openmicroscopy/ome-documentation/pull/1830/files#diff-f7e55dec15373a4b0fe85781b7e29929R47.
I think we could stick to what Webclient displays, which is
First name
Last name
Username
Password

File: omero/index.rst
Comment (hflynn) [reply omero/index.rst:36]: It is yes, should be fine once the release is live, I checked the conf.py just now

File: omero/sysadmins/public.rst
Comment (kennethgillen) [reply omero/sysadmins/public.rst:152]: @joshmoore - can we link ["launching web viewer"](https://github.com/openmicroscopy/ome-documentation/pull/1830/files#diff-f7e55dec15373a4b0fe85781b7e29929R13) back to this page?

**Comment by hflynn** [comment]
Updated https://github.com/ome/blog/pull/50 to reflect the new URL for the workflow, will update the website PR next.

File: omero/sysadmins/public.rst
Comment (pwalczysko) [inline omero/sysadmins/public.rst]: ``...selected added (or removed)...`` ?
Maybe ``to be added (or removed)`` or at least put comma between ``selected`` and ``added`` ?

Comment (joshmoore) [reply omero/sysadmins/public.rst]: I think I meant "selectively". Pushing that now.

File: omero/sysadmins/public.rst
Comment (pwalczysko) [inline omero/sysadmins/public.rst]: Unfortunately, this reads now ``Configure the Configuring public user.`` when built

File: omero/sysadmins/index.rst
Comment (pwalczysko) [reply omero/sysadmins/index.rst:146]: I guess the attack on the "Complete workflow" words is the easiest and would sort 90% of what I mean. I am afraid dispersing the content completely would create problems for simple linking from the blog post.

``Full example of ... `` would sound best to me

File: omero/sysadmins/public.rst
Comment (joshmoore) [reply omero/sysadmins/public.rst:152]: Does the link at the top count? (see https://github.com/openmicroscopy/ome-documentation/pull/1830/commits/c6cd42c9ae2602e4a2279522e86c6f346faeca2b)

File: omero/sysadmins/public.rst
Comment (will-moore) [inline omero/sysadmins/public.rst]: It might be better to discuss the logic / decision-making for setting up a second web server as a first step *before* the config steps. Currently it looks like you can make this decision at point 3, after doing steps 1 and 2.

E.g.
"Configuring OMERO.web

If you wish to have an automatically logged-in public user while still giving your existing OMERO users an unchanged user experience (i.e. not automatically logging them in as the public user), a dedicated, separate web server for servicing the public workflows can be added and configured to point at your existing OMERO.server. 
This is the workflow we adopted by adding a public OMERO.web at omero.lifesci.dundee.ac.uk, without changing the existing OMERO.web at nightshade.openmicroscopy.org.

#. Create the public user as described above. 
#. Configure the public user settings for OMERO.web, setting the :property:`omero.web.public.url_filter` to allow access to the full webclient UI as :ref:`described <public.url_filter>`.

File: omero/sysadmins/public.rst
Comment (kennethgillen) [reply omero/sysadmins/public.rst:152]: A "Ref" with link directly after [the sentence "please remember"](https://github.com/openmicroscopy/ome-documentation/commit/c6cd42c9ae2602e4a2279522e86c6f346faeca2b#diff-b0666ac64875d77a655a2a545b172593R75) would be nicer imho.

File: omero/sysadmins/public.rst
Comment (will-moore) [reply omero/sysadmins/public.rst:159]: Maybe just add that info to the repo's README, instead of here.

File: omero/sysadmins/whatsnew.rst
Comment (pwalczysko) [inline omero/sysadmins/whatsnew.rst]: As there is no summary list of links on the ``Publishing data using OMERO.web`` page (unlike on the other pages to which the What's new points) I would add some hint about how to look for it on the page. Something like the name of the paragraph of the walkthrough itself. This will of course depend on how the paragraph will be named as discussed in https://github.com/openmicroscopy/ome-documentation/pull/1830/files#r162590029

**Comment by pwalczysko** [comment]
Finished reviewing. Couple of minor comments.

File: omero/sysadmins/public.rst
Comment (joshmoore) [reply omero/sysadmins/public.rst:152]: https://github.com/openmicroscopy/ome-documentation/pull/1830/commits/f9329233b51603d5d47d5bd8e407a1b937d81a14 ?

File: omero/sysadmins/public.rst
Comment (joshmoore) [reply omero/sysadmins/public.rst:159]: See https://github.com/openmicroscopy/prod-playbooks/pull/30

File: omero/sysadmins/public.rst
Comment (joshmoore) [reply omero/sysadmins/public.rst]: Pushed

**Comment by joshmoore** [comment]
> Not sure about the capitalization or not

@pwalczysko : pushed

File: omero/sysadmins/public.rst
Comment (joshmoore) [reply omero/sysadmins/public.rst]: Pushed

File: omero/sysadmins/index.rst
Comment (joshmoore) [reply omero/sysadmins/index.rst:146]: Pushed

File: omero/sysadmins/public.rst
Comment (manics) [reply omero/sysadmins/public.rst:159]: Just in case you weren't aware a lot of the important config (e.g. omero.web.public config) is currently in private group_vars.

Comment (joshmoore) [reply omero/sysadmins/public.rst:159]: Yeah, I'm realizing having tried to get demo running in molecule:

```
+ansible:
+  playbook: playbook.yml
+  host_vars:
+    ome-demoserver.openmicroscopy.org:
+      # Override due to testing environment
+      omero.jvmcfg.system_memory: 2048
+      # Variables which are assumed to exist (missing defaults)
+      omero_server_mail_from: root@localhost
+      omero_server_mail_host: localhost
+      omero_web_admins: [[ "Sysadmin", "root@localhost"]]
+      nginx_ssl_files_path: /etc/nginx/ssl
+      vault:
+        omero_server_db_user: omero
+        omero_server_dbpassword: password
+        omero_server_dbname: omero
+        omero_server_rootpassword: password
+        omero_web_public_user: public_user
+        omero_web_public_password: password
```

I'm going to propose that we put defaults for all of these in the repo and only override what's truly secret.

Comment (manics) [reply omero/sysadmins/public.rst:159]: This is what I've done for the IDR:
- default/secret password: https://github.com/IDR/deployment/blob/IDR-0.4.4/ansible/group_vars/omero-hosts.yml#L123
- https cert location: https://github.com/IDR/deployment/blob/IDR-0.4.4/ansible/group_vars/proxy-hosts.yml#L15
- https self-signed control https://github.com/IDR/deployment/blob/IDR-0.4.4/ansible/idr-proxy.yml#L87

**Comment by hflynn** [comment]
This all looks good to me, thanks very much. Having the subheadings for this page on the sysadmin index page is definitely valuable and the reworked page flows nicely.

I'm happy to merge if there are no further comments. Are you happy now @pwalczysko ?

**Review by hflynn — APPROVED** [review]

**Comment by will-moore** [comment]
Looks good, thanks.

**Comment by pwalczysko** [comment]
Yes, it reads well. All problems fixed. Thank you.

**Comment by joshmoore** [comment]
Thanks everyone.

File: omero/sysadmins/public.rst
Comment (kennethgillen) [inline omero/sysadmins/public.rst:269]: @joshmoore  - worth linking to our example at https://github.com/openmicroscopy/prod-playbooks/blob/master/nightshade-webclients.yml#L172 ?

**Comment by hflynn** [comment]
I'll open a PR for Kenny's suggestion now

**Pull request merged**
